scripts/get_deployed_brackets.js

```diff
@@ -0,0 +1,65 @@
+const createCsvWriter = require("csv-writer").createObjectCsvWriter
+const exchangeUtils = require("@gnosis.pm/dex-contracts")
+const { fetchTokenInfoFromExchange, getExchange, getDeployedBrackets } = require("./utils/trading_strategy_helpers")(
+  web3,
+  artifacts
+)
+
+const argv = require("./utils/default_yargs")
+  .option("masterSafe", {
+    type: "string",
+    describe: "Address of Gnosis Safe owning every bracket",
+    demandOption: true,
+  })
+  .option("outputFile", {
+    type: "string",
+    describe: "Path and file name for the output of the script",
+    demandOption: false,
+  }).argv
+
+module.exports = async (callback) => {
+  try {
+    const bracketAddresses = await getDeployedBrackets(argv.masterSafe)
+
+    // writing the brackets into a csv file
+    const csvWriter = createCsvWriter({
+      path: argv.outputFile || "./bracket-addresses.csv",
+      header: [
+        { id: "Address", title: "Address" },
+        { id: "Type", title: "Type" },
+        { id: "Description", title: "Description" },
+        { id: "Tags", title: "Tags" },
+        { id: "EL", title: "Etherscan link" },
+        { id: "CP", title: "Easy C&P for JSON/SQL" },
+      ],
+    })
+
+    const exchange = await getExchange(web3)
+    const records = await Promise.all(
+      bracketAddresses.map(async (bracketAddress) => {
+        const orders = exchangeUtils.decodeOrdersBN(await exchange.getEncodedUserOrders.call(bracketAddress))
+        let tradingPair
+        if (orders.length > 0) {
+          const tokenInfoPromises = fetchTokenInfoFromExchange(exchange, [orders[0].buyToken, orders[0].sellToken])
```

Comment:
Not from this PR, but I thought the signature (params and return value) of `fetchTokenInfoFromExchange` was a bit strange when I read it. If this is working no need to change.

Also, ERC20 doesn't necessarily implement symbol or decimals, and it looks like if the contract fails to get the script would fail. So probably a token like SAI o MKR wouldn't work.

Reply:
Good point. What's the standard if the token has no decimals when printing amounts? It defaults to 18?

Reply:
It's a good point, we will address it in another PR.

Reply:
Regarding decimals or name, I don't think there's an official "default"
I think some people use 18 as the default, but I don't think is part of the standard: https://eips.ethereum.org/EIPS/eip-20

Regarding the method signature, since it's a script, I wouldn't bother too much, but imo since we are discussing it 😀, if it's an async call it should return a Promise, not an object with ids as keys and promises as values. That's what I find strange.

If the reason is, to allow parallel processing, I would define it like `fetchTokenInfoFromExchange({ exchangeArtifiact, tokenId }): Promise<TokenInfo>`, so you can easily implement any async with that.

nit: Inside the implementation, it uses a `for`, but normally using a map is more common practice. --> map a list of Ids to a list of promises 

nit: It uses global const for caching, that normally should be using UPER_CASE and placed on top. 

nit: Any non-trivial params like this one should have named params, to make it clear what it receives and prevent errors. 

Recommended also to use typescrypt, to prevent issues